perun-engine/src/test/java/cz/metacentrum/perun/engine/unit/SendCollectorTest.java

```diff
@@ -0,0 +1,63 @@
+package cz.metacentrum.perun.engine.unit;
+
+import cz.metacentrum.perun.engine.AbstractEngineTest;
+import cz.metacentrum.perun.engine.exceptions.TaskExecutionException;
+import cz.metacentrum.perun.engine.runners.SendCollector;
+import cz.metacentrum.perun.engine.scheduling.impl.BlockingSendExecutorCompletionService;
+import cz.metacentrum.perun.taskslib.model.Task;
+import cz.metacentrum.perun.taskslib.model.TaskResult;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Date;
+
+import static cz.metacentrum.perun.taskslib.model.SendTask.SendTaskStatus.ERROR;
+import static cz.metacentrum.perun.taskslib.model.SendTask.SendTaskStatus.SENT;
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+
+public class SendCollectorTest extends AbstractEngineTest {
+	private BlockingSendExecutorCompletionService sendCompletionServiceMock;
+	private SendCollector spy;
+
+	@Before
+	public void setUp() throws Exception {
+		super.mockSetUp();
+		sendCompletionServiceMock = mock(BlockingSendExecutorCompletionService.class);
+		SendCollector sendCollector = new SendCollector(sendCompletionServiceMock,
+				schedulingPoolMock, jmsQueueManagerMock);
+		spy = spy(sendCollector);
+	}
+
+	@Test
+	public void sendCollectorTest() throws Exception {
+		when(sendCompletionServiceMock.blockingTake()).thenReturn(sendTask1, sendTask2, sendTask4, sendTask3);
+		doReturn(false, false, false, false, true).when(spy).shouldStop();
+
+		spy.run();
+
+		assertEquals(SENT, sendTask1.getStatus());
+		assertEquals(SENT, sendTask2.getStatus());
+		assertEquals(SENT, sendTask3.getStatus());
+		assertEquals(SENT, sendTask4.getStatus());
+		verify(schedulingPoolMock, times(4)).decreaseSendTaskCount(task1.getId(), 1);
```

Comment:
This fails the test now. You changed implementation to call ``removeSendTaskFuture()`` instead in ``SendCollector``, which internally calls ``decreaseSendTaskCount()``, but i guess it's not recognized by the mock object in test. Something like this:
```java
verify(schedulingPoolMock, times(1)).removeSendTaskFuture(task1.getId(), sendTask1.getDestination());
```
should work, just add it for each ``sendTask`` variable from the test.